src/plugins/bellman_functions.jl

```diff
@@ -650,36 +656,40 @@ function write_cuts_to_file(
         )
         oracle = node.bellman_function.global_theta
         for (cut, state) in zip(oracle.cuts, oracle.sampled_states)
-            intercept = cut.intercept
-            for (key, π) in cut.coefficients
-                intercept += π * state.state[key]
-            end
-            push!(
-                node_cuts["single_cuts"],
-                Dict(
-                    "intercept" => intercept,
-                    "coefficients" => copy(cut.coefficients),
-                    "state" => copy(state.state),
-                ),
-            )
-        end
-        for (i, theta) in enumerate(node.bellman_function.local_thetas)
-            for (cut, state) in zip(theta.cuts, theta.sampled_states)
+            if !write_only_selected_cuts || cut.constraint_ref !== nothing
                 intercept = cut.intercept
                 for (key, π) in cut.coefficients
                     intercept += π * state.state[key]
                 end
                 push!(
-                    node_cuts["multi_cuts"],
+                    node_cuts["single_cuts"],
                     Dict(
-                        "realization" => i,
                         "intercept" => intercept,
                         "coefficients" => copy(cut.coefficients),
                         "state" => copy(state.state),
                     ),
                 )
             end
         end
+        for (i, theta) in enumerate(node.bellman_function.local_thetas)
+            for (cut, state) in zip(theta.cuts, theta.sampled_states)
+                if !write_only_selected_cuts || cut.constraint_ref !== nothing
```

Comment:
How about we keep the old indentation, and we do instead:
```suggestion
                if write_only_selected_cuts && cut.constraint_ref === nothing
                    continue
                end
```
I prefer fewer levels of indentation and early return/continue/break where possible.